src/main/java/de/tum/cit/aet/artemis/lti/web/LtiResource.java

```diff
@@ -121,32 +123,62 @@ public ResponseEntity<OnlineCourseConfiguration> updateOnlineCourseConfiguration
     }
 
     /**
-     * Handles the HTTP POST request for LTI 1.3 Deep Linking. This endpoint is used for deep linking of LTI links
-     * for exercises within a course. The method populates content items with the provided course and exercise identifiers,
-     * builds a deep linking response, and returns the target link URI in a JSON object.
+     * Handles the HTTP POST request for LTI 1.3 Deep Linking. This endpoint enables deep linking of LTI links
+     * for various course-related resources, such as exercises, lectures, competencies, learning paths, and IRIS integrations.
+     * The method processes the provided course and resource identifiers, validates LTI configurations,
+     * and constructs a deep linking response based on the specified resource type. The resulting deep link target URI
+     * is returned in a JSON object.
      *
-     * @param courseId             The identifier of the course for which the deep linking is being performed.
-     * @param exerciseIds          The identifier of the exercises to be included in the deep linking response.
-     * @param ltiIdToken           The token holding the deep linking information.
-     * @param clientRegistrationId The identifier online of the course configuration.
+     * @param courseId             The identifier of the course for which deep linking is being performed.
+     * @param exerciseIds          A set of exercise identifiers to be included in the deep linking response (optional).
+     * @param lectureIds           A set of lecture identifiers to be included in the deep linking response (optional).
+     * @param competency           A flag indicating whether the deep linking request is for a competency resource.
+     * @param learningPath         A flag indicating whether the deep linking request is for a learning path resource.
+     * @param iris                 A flag indicating whether the deep linking request is for an IRIS integration.
+     * @param ltiIdToken           The token containing the deep linking information.
+     * @param clientRegistrationId The identifier of the course's online configuration.
      * @return A ResponseEntity containing a JSON object with the 'targetLinkUri' property set to the deep linking response target link.
+     * @throws ParseException           If the LTI ID token cannot be parsed.
+     * @throws BadRequestAlertException If LTI is not configured for the course or if no valid deep linking type is provided.
      */
     @PostMapping("lti13/deep-linking/{courseId}")
-    @EnforceAtLeastInstructor
-    public ResponseEntity<String> lti13DeepLinking(@PathVariable Long courseId, @RequestParam(name = "exerciseIds") Set<Long> exerciseIds,
+    @EnforceAtLeastInstructorInCourse
+    public ResponseEntity<String> lti13DeepLinking(@PathVariable Long courseId, @RequestParam(name = "exerciseIds", required = false) Set<Long> exerciseIds,
```

Comment:
I suggest to change the input parameters. It seems that all options are mutually exclusive so you could directly send a required type (potentially string or if possible and enum) whether exercises, lectures, competency, iris or learningPath is linked. Then you have a second parameter contentIds (required = false) which are either exerciseIds or lectureIds based the previously defined type. This would be much cleaner and would also allow you to improve the code below by e.g. using a switch expression